### PR TITLE
kubeadm: clean up unnecessary references to UnknownCRISocket

### PR DESCRIPTION
**Pull request by SataQiu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Since #120828 has been merged, we are able to control whether to perform CRI detection by parameters, and pre-fill the `CRISocket` field of `InitConfiguration` is no longer needed.

This PR tries to clean up unnecessary references to `constants.UnknownCRISocket`.

#### Which issue(s) this PR fixes:

Ref #117614 #97625

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by SataQiu** [comment]
/hold 
for review

**Review by pacoxu — APPROVED** [review]
/lgtm
this can be removed IMO.

File: cmd/kubeadm/app/cmd/certs.go
Comment (neolit123) [inline cmd/kubeadm/app/cmd/certs.go:137]: are we sure this (and similar changes) will not cause the autodetection to start happening for commands that don't need it? 
we might be lacking full unit test coverage around the cri socket detection problem.

Comment (SataQiu) [reply cmd/kubeadm/app/cmd/certs.go:137]: > we might be lacking full unit test coverage around the cri socket detection problem.

@neolit123 To be honest, yes. 
Currently, we have to check it manually. That's what I've done.

For this change, as we have passed the `skipCRIDetect = true` parameter to `configutil.LoadOrDefaultInitConfiguration()`, the auto-detection will not happen. 
And the other changes are similar.
https://github.com/kubernetes/kubernetes/blob/c3bf541edeefc1cfcf24ef5e4cb5481e529c78f8/cmd/kubeadm/app/cmd/certs.go#L135-L140

Comment (neolit123) [reply cmd/kubeadm/app/cmd/certs.go:137]: ok, LGTM.
 i leave this to your judgement.

**Pull request merged**
